WordPress/Classes/ViewRelated/System/Search/WPSearchControllerConfigurator.m

```diff
@@ -0,0 +1,112 @@
+#import "WPSearchControllerConfigurator.h"
+#import "WordPress-Swift.h"
+
+const CGFloat SearchBarWidth = 280.0;
+const CGFloat SearchBariPadWidth = 600.0;
+const CGFloat SearchWrapperViewPortraitHeight = 64.0;
+const CGFloat SearchWrapperViewLandscapeHeight = 44.0;
+const NSTimeInterval SearchBarAnimationDuration = 0.2; // seconds
+
+@interface WPSearchControllerConfigurator ()
+
+@property (nonatomic, weak) WPSearchController *searchController;
+@property (nonatomic, weak) UIView *searchWrapperView;
+@property (nonatomic, weak) UISearchBar *searchBar;
+
+@end
+
+@implementation WPSearchControllerConfigurator
+
+- (instancetype)initWithSearchController:(WPSearchController *)searchController
+                   withSearchWrapperView:(UIView *)searchWrapperView
+{
+    self = [super init];
+    if (self) {
+        _searchController = searchController;
+        _searchWrapperView = searchWrapperView;
+        _searchBar = _searchController.searchBar;
+    }
+    
+    return self;
+}
+
+- (void)configureSearchControllerBarAndWrapperViewOfClass:(Class)class
```

Comment:
I think this method name is confusing.  The reason is that you're using the term `WrapperView` to refer to the parent VCs (in both cases where this method is called a VC is passed).  The term `WrapperView` is also present in the property `searchWrapperView` above, making it a bit confusing to the reader what's being passed here.

Also a more appropriate parameter type would be `Class <UIAppearanceContainer>`, since that's what `appearanceWhenContainedIn` receives as input parameters.

My suggestion is to change this to something along the lines of:

`- (void)configureControllerWithParentAppearanceContainer:(Class <UIAppearanceContainer>)class;`